keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1813 @@
+        pod-level limit is applied to each container's cgroup maximum value. This
+        is because container-level limit is implied in this case, and some
+        runtimes (like the Java runtime) rely on container-level cgroup maximum
+        values for fine tuning their components. This implied pod-level limit is
+        not set in Pod spec.  
+
+  * **Rows 9 - 12**
+
+    ![](./pod-request-set.png)
+
+      * When the pod-level limit is unset, the value will be derived from
+        container-level limits. If no pod-level limit is defined and not all the
+        containers within the pod have explicitly limits set, then the containers in the pod can
+        use full amount of resources available on the node, up to the node's
+        allocatable capacity.
+
+      * Row 10 Annotation [b]: For container level resources, if a request is not
+        specified but a limit it set, the Pod Spec is modified to set the [request
+        equal to limit](https://github.com/kubernetes/kubernetes/blob/005f184ab631e52195ed6d129969ff3914d51c98/pkg/apis/core/v1/defaults.go#L167-L228) for each container. This modified Pod spec is then used by
+        the scheduler, and also for configuring cgroup settings.  
+
+  * **Rows 13 - 16**
+
+    ![](./pod-request-limit-set.png)
+
+      * When both pod-level request and limit are explicitly specified,the scheduler and
+        cgroups will use those values directly.
+    
+      * Rows 13, 15 Annotation [c]: When container-level limit is not set, the
+        pod-level limit is applied to each container's cgroup maximum value. This
+        is because container-level limit is implied in this case, and some
+        runtimes (like the Java runtime) rely on container-level cgroup maximum
+        values for fine tuning their components.
+<br>
+
+##### Clarifying Tricky Cases and Design Decisions with Pod Level Resources
+
+1. [Invalid Pod Spec] Row 5 from the scenarios table: Pod Limits Defined; Pod
+   Requests & Container Specs Unspecified.
+   
+   ![](./example-1.png)
+  
+   * The Challenge of Missing Requests: In this case, Kubernetes cannot determine
+     the appropriate pod-level request as pod-level request is not set explicitly,
+     and since container-level requests are also not set, there's no way for
+     kubernetes to derive pod-level requests from the resource spec.
+   * Prioritizing Explicit Resource Requests: Instead of relying on potentially
+     inaccurate default values or guessing the user's intent, this KEP proposes
+     rejecting such a pod specification. This encourages users to explicitly
+     define their resource requests, promoting clarity and preventing ambiguity in
+     resource allocation.  
+
+2. [Invalid Pod Spec] Row 6 from the scenarios table: Pod Limits and Container
+   Limits Defined, Pod Requests and Container Requests Unspecified.
+
+   ![](./example-2.png)
+
+   * This pod will be rejected for same reasons as example 1 above (for row 5).
```

Comment:
I think this one has a far more obvious reason to fail:

* The Container requests are defaulted from limit (60, 60)
* The aggregate request (120) is greater than the pod's limit

Reply:
Correct